bundles/sirix-core/src/main/java/org/sirix/access/trx/node/AbstractNodeHashing.java

```diff
@@ -158,7 +158,7 @@ private void postorderAdd() {
       cursorToRoot = pageTrx.prepareRecordForModification(getCurrentNode().getNodeKey(), IndexType.DOCUMENT, -1);
       hashCodeForParent = getCurrentNode().computeHash().add(hashCodeForParent.multiply(PRIME));
       // Caring about attributes and namespaces if node is an element.
-      if (cursorToRoot.getKind() == NodeKind.ELEMENT) {
+      if (cursorToRoot.getPathKind() == NodeKind.ELEMENT) {
```

Comment:
this should be the `getKind()`-method, but somewhere I think you changed something between `getKind()` and `getPathKind()`. But over here it checks the type of the node itself, not the path type/kind.

Reply:
Accordingly all the other changes in this commit :)

Reply:
ok, I will revert the getKind and getPathKind that I changed on main/java/org/sirix/index/path/summary/PathNode.java for now, but, just for you know sonar is complaining as a critical bug. https://sonarcloud.io/project/issues?id=sirixdb_sirix&open=AXy4_zUSQgQXZ6uows07&resolved=false&severities=CRITICAL&types=BUG

Reply:
Done :) if anything needs to be changed please let me know, after this gets merged, I will continue working on more sonar issues.

Reply:
getKind() should return NodeKind.PATH as that's the type of the PathNode. The `getPathKind()`-method should return which type of a path node the current node denotes (for JSON: array or object fields, for XML: attribute or element).

That is a path node is a node of a path summary, which simply is a "small" tree, which denotes all paths in a resource :-) Hope that makes sense and helps in getting the context. Did you already push your revert?

Reply:
Oh I see, so we might need to add the @ignore tag for sonar on this class and/or might be useful to add a comment on it, as Sonar thinks this is an issue. yes, changes on PathNode class are reverted! :)

Reply:
Yes, I've closed the issue on Sonar with a short description. We have to be careful about false positives.

I can't find the reverted changes. `AbstractNodeHashing` for instance still uses `getPathKind()` as far as I can see.

Reply:
it got messy I will commit again in another PR.

Reply:
Thanks and looking forward to your contributions :-)